plugins/extractors/googleworkspace/googleworkspace.go

```diff
@@ -0,0 +1,214 @@
+    "private_key": "-----BEGIN PRIVATE KEY-----\njbjabdjbajd\n-----END PRIVATE KEY-----\n",
+    "client_email": "meteor-sa@odpf-project.iam.gserviceaccount.com",
+    "client_id": "1100599572858548635286",
+    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+    "token_uri": "https://oauth2.googleapis.com/token",
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/meteor-sa%40odpf-project.iam.gserviceaccount.com"
+}
+user_email: user@odpf.com`
+
+var info = plugins.Info{
+	Description:  "User list from Google Workspace",
+	SampleConfig: sampleConfig,
+	Tags:         []string{"platform", "extractor"},
+}
+
+// Extractor manages the extraction of data from the extractor
+type Extractor struct {
+	plugins.BaseExtractor
+	logger      log.Logger
+	config      Config
+	TokenSource oauth2.TokenSource
+	Client      *admin.Service
+	emit        plugins.Emit
+}
+
+// New returns a pointer to an initialized Extractor Object
+func New(logger log.Logger) *Extractor {
+	e := &Extractor{
+		logger: logger,
+	}
+	e.BaseExtractor = plugins.NewBaseExtractor(info, &e.config)
+
+	return e
+}
+
+// Init initializes the extractor
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	if err = e.BaseExtractor.Init(ctx, config); err != nil {
+		return err
+	}
+
+	jwtConfig, err := google.JWTConfigFromJSON([]byte(e.config.ServiceAccountJSON), admin.AdminDirectoryUserScope)
+	if err != nil {
+		return plugins.InvalidConfigError{}
+	}
+
+	if e.config.UserEmail == "" {
+		return plugins.InvalidConfigError{}
+	}
+	jwtConfig.Subject = e.config.UserEmail
+	ts := jwtConfig.TokenSource(ctx)
+
+	e.TokenSource = ts
+
+	return
+}
+
+// Extract extracts the data from the extractor
+// The data is returned as a list of assets.Asset
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error) {
+	var status string
+	e.emit = emit
+	r, err := FetchUsers(ctx, e.TokenSource)
+	if err != nil {
+		return err
+	}
+
+	if len(r.Users) == 0 {
+		e.logger.Info("No users found.\n")
+	} else {
+		for _, u := range r.Users {
+			if !u.Suspended {
+				status = "not suspended"
+			} else {
+				status = "suspended"
+			}
+
+			var userAttributes = make(map[string]interface{})
+			userAttributes = buildOrgAttributes(userAttributes, u.Organizations)
+			userAttributes = buildRelationsAttributes(userAttributes, u.Relations)
+			userAttributes = buildCustomSchemasAttributes(userAttributes, u.CustomSchemas)
+			userAttributes["org_unit_path"] = u.OrgUnitPath
+			userAttributes["aliases"] = strings.Join(u.Aliases, ",")
+
+			e.emit(models.NewRecord(&assetsv1beta1.User{
+				Resource: &commonv1beta1.Resource{
+					Service: "google workspace",
+					Name:    u.Name.FullName,
+					Urn:     u.PrimaryEmail,
+				},
+				Email:    u.PrimaryEmail,
+				FullName: u.Name.FullName,
+				LastName: u.Name.FamilyName,
```

Comment:
`FullName` and a `LastName` 
Do we need a `FirstName` as well? 
**Or**
Do we need `LastName` when we have the `FullName` already

Reply:
Yes my bad, will remove the LastName.